flytecopilot/data/download.go

```diff
@@ -31,40 +33,120 @@ type Downloader struct {
 	mode core.IOStrategy_DownloadMode
 }
 
-// TODO add support for multipart blobs
-func (d Downloader) handleBlob(ctx context.Context, blob *core.Blob, toFilePath string) (interface{}, error) {
-	ref := storage.DataReference(blob.Uri)
-	scheme, _, _, err := ref.Split()
+func (d Downloader) handleBlob(ctx context.Context, blob *core.Blob, toPath string) (interface{}, error) {
+	blobRef := storage.DataReference(blob.Uri)
```

Comment:
```suggestion
func (d Downloader) handleBlob(ctx context.Context, blob *core.Blob, toPath string) (interface{}, error) {
	blobRef := blob.Uri
```

Reply:
blob.Uri is string type, if we don't use DataReference to wrap it, we can't split it (next line), did I misunderstand?

Reply:
`storage.DataReference` is also a str type
I mean you are doing this `string(blob.uri)`, but you can do this `blob.uri` directly.